action/protocol/staking/read_state.go

```diff
@@ -79,6 +79,16 @@ func readStateBucketsByCandidate(ctx context.Context, sr protocol.StateReader, c
 	return toIoTeXTypesVoteBucketList(buckets)
 }
 
+func readStateBucketByIndex(ctx context.Context, sr protocol.StateReader,
```

Comment:
not sure what's the major use case for this.
if we expect user has a list of indexes to query. a `bucketsByIndexes` is better which reduce number of api calls